packages/web3-utils/src/socket_provider.ts

```diff
@@ -331,6 +349,27 @@ export abstract class SocketProvider<
 		this._onDisconnect(disconnectCode, data);
 	}
 
+	/**
+	 * Safely disconnects the socket, async and waits for request size to be 0 before disconnecting
+	 * @param code - The code to be sent to the server
+	 * @param data - The data to be sent to the server
+	 */
+	public async safeDisconnect(code?: number, data?: string) {
+		const checkQueue = async () => 
+			new Promise(resolve => {
+				const interval = setInterval(() => {
```

Comment:
If the user has a logic to force the disconnect, if not the safe disconnect finished within a time, there is no way for the user to clear this interval. For this, I suggest to make `_clearQueues` as public. So the user can call if he wanted to force the disconnect after sometime pass.

We can also imbed this logic of force close after a selected time within this method, as an optional param.

Reply:
thanks for the suggestions, added them